stages/org.osbuild.zip

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python3
+"""
+Compresses an input file using zip.
+"""
+
+import os
+import subprocess
+import sys
+
+import osbuild.api
+
+CAPABILITIES = ["CAP_MAC_ADMIN"]
+
+
+SCHEMA_2 = """
+"options": {
+  "additionalProperties": false,
+  "required": ["filename"],
+  "properties": {
+    "filename": {
+      "description": "Filename for tar archive",
+      "type": "string"
+    },
+    "paths": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "description": "If specified, the archive will only contain the specified paths, in the specified order"
+      }
+    },
+    "level": {
+      "description": "Compression level",
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 9,
+      "default": 6
+    }
+  }
+},
+"inputs": {
+  "type": "object",
+  "additionalProperties": false,
+  "required": ["tree"],
+  "properties": {
+    "tree": {
+      "type": "object",
+      "additionalProperties": true
+    }
+  }
+}
+"""
+
+
+def main(inputs, output_dir, options):
+    tree = inputs["tree"]["path"]
+    filename = options["filename"].lstrip("/")
+    level = options.get("level", 6)
+    paths = options.get("paths", [])
+
+    extra_args = []
+
+    # Set up the zip command.
+    zip_cmd = [
+        "zip",
+        "-r",
+        f"-{level}",
+        *extra_args,
+        os.path.join(output_dir, filename),
+        "."
+    ]
+
+    if paths:
+        zip_cmd += ["-i"]
```

Comment:
Does it work? Are you not getting something like `-ipath` instead of `-i path`, seems you need a space here?

Reply:
Since `-i` is an element within a list, it gets appended as another element to the `zip_cmd` list, which separates it by a comma. For example:

```python
zip_cmd = ["zip", "-r", "/some/dir"] # ["zip", "-r", "/some/dir"]
zip_cmd += ["-i"] # ["zip", "-r", "/some/dir", "-i"]
paths = ["/path/a", "/path/b"]
zip_cmd += paths # ["zip", "-r", "/some/dir", "-i", "/path/a", "/path/b"]
```
With the way that `subprocess.run` works, it takes an input command as a list of strings, meaning that spaces are not required.

Reply:
yeah it feels weird, but you only have to specify `-i` once and then a space separate list of files on the bash command line. so this looks like it works